### PR TITLE
PUPIL-536 - Feat/404 sensitive lessons

### PR DESCRIPTION
**Pull request by benprotheroe**
## Description

Music year: 2012

- List of changes
query lessons with isSensitive field
404 any lesson which is sensitive

## Issue(s)

Fixes #

## How to test

1. Go to [abortion page](https://deploy-preview-2331--oak-web-application.netlify.thenational.academy/pupils/programmes/rshe-pshe-secondary-ks4-l/units/intimate-sexual-relationships-pregnancy-and-parenting-ff99/lessons/abortion-6xh38d)
2. [online world vs real life page](https://deploy-preview-2331--oak-web-application.netlify.thenational.academy/pupils/lessons/how-the-online-world-is-different-to-real-life-c8vk4d)
3. Click on \_\_\_\_\_\_
4. You should see \_\_\_\_\_\_

## Screenshots

How it used to look (delete if n/a):
{screenshots}

How it should now look:
{screenshots}

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/node-lib/curriculum-api-2023/queries/pupilLessonOverview/pupilLessonOverview.query.ts [hunk 1]
```diff
@@ -18,6 +18,10 @@ export const pupilLessonOverviewQuery =
       throw new OakError({ code: "curriculum-api/not-found" });
     }
 
+    if (lesson.isSensitive === true) {
+      throw new OakError({ code: "curriculum-api/not-found" });
+    }
+
```
Comment (carlmw) [inline src/node-lib/curriculum-api-2023/queries/pupilLessonOverview/pupilLessonOverview.query.ts:24]:
```suggestion
    if (lesson.isSensitive) {
      throw new OakError({ code: "curriculum-api/not-found" });
    }

```

🏄🏻 the equality check is redundant with a boolean

File: src/node-lib/curriculum-api-2023/queries/pupilLessonOverviewCanonical/pupilLessonOverviewCanonical.query.ts [hunk 2]
```diff
@@ -11,6 +11,10 @@ export const pupilLessonOverviewCanonicalQuery =
       throw new OakError({ code: "curriculum-api/not-found" });
     }
 
+    if (lessons[0]?.isSensitive === true) {
+      throw new OakError({ code: "curriculum-api/not-found" });
+    }
```
Comment (carlmw) [inline src/node-lib/curriculum-api-2023/queries/pupilLessonOverviewCanonical/pupilLessonOverviewCanonical.query.ts:16]:
```suggestion
    if (lessons[0]?.isSensitive) {
      throw new OakError({ code: "curriculum-api/not-found" });
    }
```

🏄🏻

**Review by carlmw — COMMENTED** [review]
Code looks good to me, but I can't seem to access any other lessons from the deploy preview. https://deploy-preview-2331--oak-web-application.netlify.thenational.academy/pupils/programmes/english-secondary-ks4-eduqas/units/writing-masters/lessons/writing-excellent-descriptions

https://deploy-preview-2331--oak-web-application.netlify.thenational.academy/pupils/lessons/prologue-grime-mix-ccuk0d

**Review by kimon-satan — APPROVED** [review]

**Comment by rightsaidjames** [comment]
Regression tested with the following (non-sensitive lessons):
https://deploy-preview-2331--oak-web-application.netlify.thenational.academy/pupils/l/the-firework-makers-daughter-by-phillip-pullman-95e6/lessons/to-engage-with-the-text-c9h3je
https://deploy-preview-2331--oak-web-application.netlify.thenational.academy/pupils/lessons/prologue-grime-mix-ccuk0d
https://deploy-preview-2331--oak-web-application.netlify.thenational.academy/pupils/l/year-7-unit-2-0ebb/lessons/extended-reading-sept-couleurs-magiques-part-12-65gpar

@benprotheroe Good to go I think. We can't test this fully until it is merged and we can test it with sensitive RSHE lessons on Migration.

**Review by rightsaidjames — APPROVED** [review]

**Review by rightsaidjames — CHANGES_REQUESTED** [review]
Sorry, correction...

It seems that this PR might have broken compatibility for 'new' lessons, for example:
programmes/maths-secondary-ks3/units/place-value/lessons/checking-understanding-of-place-value-in-integers
https://deploy-preview-2331--oak-web-application.netlify.thenational.academy/pupils/programmes/english-secondary-ks4-eduqas/units/writing-masters/lessons/writing-excellent-descriptions

@benprotheroe Can you look into this?

**Pull request merged**